dhis-2/dhis-api/src/main/java/org/hisp/dhis/schema/Property.java

```diff
@@ -259,144 +263,83 @@ public PropertyType getPropertyType() {
-  }
-
-  public void setGetterMethod(Method getterMethod) {
-    this.getterMethod = getterMethod;
-  }
-
-  public Method getSetterMethod() {
-    return setterMethod;
-  }
-
-  public void setSetterMethod(Method setterMethod) {
-    this.setterMethod = setterMethod;
-  }
-
   @JsonProperty
   @JacksonXmlProperty(namespace = DxfNamespaces.DXF_2_0)
   public String getName() {
     return name;
   }
 
-  public void setName(String name) {
-    this.name = name;
-  }
-
   @JsonProperty
   @JacksonXmlProperty(namespace = DxfNamespaces.DXF_2_0)
   public String getFieldName() {
     return fieldName;
   }
 
-  public void setFieldName(String fieldName) {
-    this.fieldName = fieldName;
-  }
-
   @JsonProperty
   @JacksonXmlProperty(namespace = DxfNamespaces.DXF_2_0)
   public boolean isPersisted() {
     return persisted;
   }
 
-  public void setPersisted(boolean persisted) {
-    this.persisted = persisted;
-  }
-
   @JsonProperty
   @JacksonXmlProperty(namespace = DxfNamespaces.DXF_2_0)
   public String getCollectionName() {
     return collectionName != null ? collectionName : (isCollection() ? name : null);
   }
 
-  public void setCollectionName(String collectionName) {
-    this.collectionName = collectionName;
-  }
-
   @JsonProperty
   @JacksonXmlProperty(namespace = DxfNamespaces.DXF_2_0)
   public Boolean isCollectionWrapping() {
     return collectionWrapping;
   }
 
-  public void setCollectionWrapping(Boolean collectionWrapping) {
-    this.collectionWrapping = collectionWrapping;
-  }
-
   @JsonProperty
   @JacksonXmlProperty(namespace = DxfNamespaces.DXF_2_0)
   public String getDescription() {
     return description;
   }
 
-  public void setDescription(String description) {
-    this.description = description;
-  }
-
   @JsonProperty
   @JacksonXmlProperty(namespace = DxfNamespaces.DXF_2_0)
   public String getNamespace() {
     return namespace;
   }
 
-  public void setNamespace(String namespace) {
-    this.namespace = namespace;
-  }
-
   @JsonProperty
   @JacksonXmlProperty(namespace = DxfNamespaces.DXF_2_0)
   public boolean isAttribute() {
     return attribute;
   }
 
-  public void setAttribute(boolean attribute) {
-    this.attribute = attribute;
-  }
-
   @JsonProperty
   @JacksonXmlProperty(namespace = DxfNamespaces.DXF_2_0)
   public boolean isSimple() {
     return simple;
   }
 
-  public void setSimple(boolean simple) {
-    this.simple = simple;
-  }
-
   @JsonProperty
   @JacksonXmlProperty(namespace = DxfNamespaces.DXF_2_0)
   public boolean isCollection() {
     return collection;
   }
 
-  public void setCollection(boolean collection) {
-    this.collection = collection;
+  @JsonProperty(access = JsonProperty.Access.READ_ONLY)
+  @JacksonXmlProperty(namespace = DxfNamespaces.DXF_2_0)
+  public boolean isSortable() {
+    Sortable sortable = getterMethod == null ? null : getterMethod.getAnnotation(Sortable.class);
+    return sortable != null ? sortable.value() : !isCollection() && isSimple();
```

Comment:
I think we need to check if the property is `persisted` , meaning it is mapped to a database column.
For example: a class may not have the property `shortName` but it is still inherit the property from base class.

Reply:
Good point, but I will have to find a way to not exclude a computed property like `displayName` that way since it isn't persisted but still possible to sort by.

Reply:
I use `persisted` now where the default is that it has to be persisted. To make properties like `displayName` still be sortable they must be annotated with `@Sortable(whenPersisted=false)`. This is not perfect but covers most of the cases.